ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -82,20 +98,42 @@ def forward(
         sequence_length: int = 1,
     ) -> Tuple[torch.Tensor, torch.Tensor]:
         encodes = []
-        for idx, processor in enumerate(self.processors):
-            obs_input = inputs[idx]
-            processed_obs = processor(obs_input)
-            encodes.append(processed_obs)
+        var_len_inputs = []  # The list of variable length inputs
 
+        for idx, processor in enumerate(self.processors):
+            if processor is not None:
+                # The input can be encoded without having to process other inputs
+                obs_input = inputs[idx]
+                processed_obs = processor(obs_input)
+                encodes.append(processed_obs)
+            else:
+                var_len_inputs.append(inputs[idx])
         if len(encodes) == 0:
+            encoded_self = torch.zeros(0, 0)
```

Comment:
I do not like this. I get that it wont actually be concatenated because if the length of encodes is 0 than the length in EntityEmbedding will be 0 and so nothing will be concatenated but this makes me uncomfortable.

Reply:
An alternative is to do None, but I get some type issues when I do.